### PR TITLE
Enable building code without System.Drawing.Common dependency

### PR DESCRIPTION
**Pull request by wieslawsoltes**
#### What does this implement/fix? Explain your changes.

Enabled building library without System.Drawing.Common dependency by defining `NO_SDC`.

### CODE REVIEW

**Comment by H1Gdev** [comment]
I don't know how to surround `NO_SDC`.
Are there any rules ?

**Comment by wieslawsoltes** [comment]
If you remove`System.Drawing.Common` package reference anything that does not compile. All the code that uses types  from System.Drawing.Common are surrounded by NO_SDC if/def.

**Comment by wieslawsoltes** [comment]
> If you remove`System.Drawing.Common` package reference anything that does not compile. All the code that uses types from System.Drawing.Common are surrounded by NO_SDC if/def.

In Svg.Skia I substituted the System.Drawing.Common with SkiaSharp alternative.

**Review by mrbean-bremen — APPROVED** [review]

**Comment by mrbean-bremen** [comment]
@wieslawsoltes - I have rebased this with master, please check if all still works with Svg.Skia and I haven't messed something up while resolving conflicts. Will merge if you give the ok.

**Comment by wieslawsoltes** [comment]
> @wieslawsoltes - I have rebased this with master, please check if all still works with Svg.Skia and I haven't messed something up while resolving conflicts. Will merge if you give the ok.

@mrbean-bremen Works fine with Svg.Skia. Thanks!

**Comment by H1Gdev** [comment]
@wieslawsoltes 

The questions are listed below.
- Where is `NO_SDC` defined ?
- How does SVG behave when `NO_SDC` is defined?

**Comment by wieslawsoltes** [comment]
> @wieslawsoltes
> 
> The questions are listed below.
> 
> * Where is `NO_SDC` defined ?
> * How does SVG behave when `NO_SDC` is defined?

In my Svg.Custom project https://github.com/wieslawsoltes/Svg.Skia/blob/407a032aeaaf9a3716959e52e63aca6953f50c6d/src/Svg.Custom/Svg.Custom.csproj#L14

When it's defined I replace all removed functionality with my code.

**Comment by H1Gdev** [comment]
@wieslawsoltes 

Thanks !

@mrbean-bremen 

This PR depends on `Svg.Skia`.
How about creating another branch ?

**Comment by mrbean-bremen** [comment]
@H1Gdev - it is the other way around, `Svg.Skia` depends on this. It is not ideal, but I see this as a first step to factor out the drawing functionality as mentioned in #590. Anyway we need a way to handle the dropped support for `Common.Drawing`, and `Svg.Skia` already does a lot here. I would like to work both repos to be able to work together more seemlessly, and this change might be a step on this way.

File: Source/Paths/SvgQuadraticCurveSegment.cs [hunk 1]
```diff
@@ -34,6 +36,7 @@ private static PointF CalculateSecondControlPoint(PointF controlPoint, PointF en
             return new PointF(x2, y2);
         }
 
+#if !NO_SDC
```
Comment (H1Gdev) [inline Source/Paths/SvgQuadraticCurveSegment.cs:39]: @wieslawsoltes 

I think it surrounds `AddToPath`.

Comment (mrbean-bremen) [reply Source/Paths/SvgQuadraticCurveSegment.cs:39]: It surrounds `AddToPath` and the helper function `CalculateControlPoint` used in `AddToPath`. Is that wrong?

Comment (wieslawsoltes) [reply Source/Paths/SvgQuadraticCurveSegment.cs:39]: I think I will submit PR with splitting of all drawing code inside NO_SDC if/def into separate files. This should be easier to read and maintain also make next step closer for https://github.com/svg-net/SVG/issues/590

Comment (wieslawsoltes) [reply Source/Paths/SvgQuadraticCurveSegment.cs:39]: Started work here https://github.com/svg-net/SVG/pull/934 @H1Gdev @mrbean-bremen your input would be appreciated

Comment (mrbean-bremen) [reply Source/Paths/SvgQuadraticCurveSegment.cs:39]: That sounds good, thanks! I will have a look later today.

**Pull request merged**